src/compiler/checker.ts

```diff
@@ -43521,6 +43551,28 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         });
     }
 
+    function checkUninitializedLocals(nodeWithLocals: HasLocals) {
+        nodeWithLocals.locals!.forEach(local => {
+            if (!(local.flags & SymbolFlags.Variable) || !local.isReferenced || isSymbolAssignedDefinitely(local)) {

```

Comment:
I think you should also check if the variable is constant, i.e. use `isConstantVariable`, otherwise we can get two errors for an uninitialized const variable, and `isSymbolAssigned` and `isSymbolAssignedDefinitely` are not intended to be used with constant variables.

Test case:
```ts
function f() {
    const x: number; // Two similar errors here
    function bar() {
        let y = x;
    }
}
```